### PR TITLE
Add SessionRequest decorator in hibernate module

### PR DESCRIPTION
**Pull request by Blquinn**
Note that the TransactionRequest now uses SessionRequest to add the session to the managed context.

ISSUE https://github.com/jooby-project/jooby/issues/1443

### CODE REVIEW

File: modules/jooby-hibernate/src/main/java/io/jooby/hibernate/TransactionalRequest.java [hunk 1]
```diff
@@ -58,43 +56,39 @@
...
-    return ctx -> {
+    return sessionRequest.apply(ctx -> {
       SessionFactory sessionFactory = ctx.require(key);
       Transaction trx = null;
       try {
-        Session session = sessionFactory.openSession();
-        ManagedSessionContext.bind(session);
+        Session session = sessionFactory.getCurrentSession();
         trx = session.getTransaction();
-        trx.begin();
```
Comment (jknack) [inline modules/jooby-hibernate/src/main/java/io/jooby/hibernate/TransactionalRequest.java]: seems wrong, why did you remove the begin trx?

File: modules/jooby-hibernate/src/main/java/io/jooby/hibernate/SessionRequest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    return (ctx) -> {
+      SessionFactory sessionFactory = ctx.require(key);
+      try {
+        Session session = sessionFactory.openSession();
+        ManagedSessionContext.bind(session);
+        Object result = next.apply(ctx);
+
+        Transaction transaction = session.getTransaction();
+        if (transaction.getStatus() == TransactionStatus.ACTIVE) {
+          transaction.rollback();
```
Comment (jknack) [inline modules/jooby-hibernate/src/main/java/io/jooby/hibernate/SessionRequest.java]: add log.error indicating this is not an expected state and shouldn't exist an active transaction

File: modules/jooby-hibernate/src/main/java/io/jooby/hibernate/TransactionalRequest.java (hunk 1 above)
Comment (Blquinn) [reply modules/jooby-hibernate/src/main/java/io/jooby/hibernate/TransactionalRequest.java]: I meant to change this to session.beginTransaction(); Which creates a transaction, begins and returns that tx.

**Comment by Blquinn** [comment]
I made those requested changes and amended the commit.

**Comment by Blquinn** [comment]
Any idea why checkstyle is failing? I see that you committed a fix for it, but still seems to fail on this branch.

**Comment by jknack** [comment]
do a build like:

```
./etc/build.sh
```

you will see the error at the console

**Comment by Blquinn** [comment]
Ok, found the error. Build is running now. :)

**Comment by imeszaros** [comment]
I really like this PR, it makes possible to implement a similar control flow to what we currently have in our Jooby 1.x projects using the `UnitOfWork` interface.
